src/overlaybd/zfile/zfile.cpp

```diff
@@ -721,11 +755,18 @@ bool load_jump_table(IFile *file, CompressionFile::HeaderTrailer *pheader_traile
         LOG_ERRNO_RETURN(0, false, "failed to read index");
     }
     ret = jump_table.build(ibuf.get(), pht->index_size,
-                     CompressionFile::HeaderTrailer::SPACE + pht->opt.dict_size,
-                     pht->opt.block_size);
+                           CompressionFile::HeaderTrailer::SPACE + pht->opt.dict_size,
+                           pht->opt.block_size, pht->opt.verify);
     if (ret != 0) {
         LOG_ERRNO_RETURN(0, false, "failed to build jump table");
     }
+    if (pht->is_digest_enabled()) {
+        LOG_INFO("check jumptable CRC32 (` expected)", pht->index_crc);
+        auto crc = crc32::crc32c(ibuf.get(), index_bytes);
+        if (crc != pht->index_crc) {
+            LOG_ERRNO_RETURN(0, false, "checksum of jumptable is incorrect");
```

Comment:
nit: might just print out the expected and actual if the string is not too long?